### DISCUSSION TITLE
sensor fusion between 2da lidar and realsense camera

### DISCUSSION THREAD

**Discussion by hardikbvaghela** [prl-mushr/mushr#79] [Q&A]
hello
i want to ask you about the sensor fusion how can i fuse both camera and lidar and get the obstacle distance and angle measurement on my screen?

**Comment by palanc** [comment]
Hi @hardikbvaghela - Here are all of our tutorials: [https://mushr.io/tutorials/](https://mushr.io/tutorials/)

And here are some that might be specifically relevant to your question:
[https://mushr.io/tutorials/realsense/](https://mushr.io/tutorials/realsense/)
[https://mushr.io/tutorials/mapping/](https://mushr.io/tutorials/mapping/)

**Reply by hardikbvaghela** [reply]
![IMG-5097](https://user-images.githubusercontent.com/93780099/166899951-79243511-5357-4662-bb76-70b9289a7fcd.jpg)
Thank you so much for your response. but I want to know if it's a 2D ydlidar. so it gives a 2d lidar point cloud. but if want to detect the pedestrian with this lidar so how can I detect it because it gives the same color point clouds surrounding the environment. so how can I detect pedestrians with this 2dlidar? I can attach an image for my Rviz. please guide me.

**Reply by schmittlema** [reply]
yes it is a 2D lidar. Lidar returns distances not classifications. The color on the pointcloud is just hardcoded for visualization. You ideally would use the camera to detect people. For example [like this](https://www.edgeimpulse.com/blog/recognizing-your-blind-spots-pedestrian-detection-system-with-nvidia-jetson-nano). Then you can use lidar (or more easily the depth image) to get the position of the person. If you want to detect people soley based on lidar there exist models for that but it is more tricky/difficult especially with 2dlidar

**Comment by schmittlema** [comment]
Hi hardikvaghela. This will take a bit of work. First you will need to classify obstacles in the image space. Most people use some neural network for this. Assuming you have a good classification using the RGB of the camera, you can use the depth image to get associated depth of the obstacles. So if some pixel is classified as a wall, the same pixel in the depth image should give you the depth of the wall. Then you can convert them to 3D points using the [following](https://stackoverflow.com/questions/31265245/extracting-3d-coordinates-given-2d-image-points-depth-map-and-camera-calibratio)

For each point in the [laserscan](http://docs.ros.org/en/noetic/api/sensor_msgs/html/msg/LaserScan.html) there is a distance and associated angle. You can convert that to x, y points via the following polar --> cartesian equations
`x = r cos(theta)`
`y = r sin(theta)`
Where r is the distance of the scan point and theta is the angle.
